### DISCUSSION TITLE
Is it possible to use family plan when self-hosting?

### DISCUSSION THREAD

**Discussion by reorx** [ente-io/ente#2623] [Q&A] [answered]
On the home page of Ente, it states:

> Sync your library with your partner, and even designate them as an heir to your account.

This is one of the features that I really appreciate, as it allows me to share my library with my partner.

However, after setting up my own Ente server, I found that the family plan does not seem to be available. My plan always remains the free 5GB plan, even after I use ente-cli to update the subscription. 

I'm wondering if there's any way to enable it, or if there's an alternative method to achieve my initial goal of sharing my entire library with another user.

**Comment by vishnukvmd** [comment, marked as answer]
Hey, if you modify the `payment-provider` attribute to anything other than `free`, family plan will become accessible.

Please note that
1. Family plan is designed to share your paid subscription, not your library. Each member gets their own private space. This feature might not be valuable in a self-hosted setup.
2. The best way to share your entire library would be to add the other user as a viewer / collaborator to all of your albums. There's a separate discussion tracking a "Share all" feature here: https://github.com/ente-io/ente/discussions/787

**Reply by reorx** [reply]
Thank you for your reply. It appears that #787 is exactly what I need.

**Reply by aqsede** [reply]
> modify the `payment-provider` attribute

Hello, how can I modify this? is it on the admin tools

**Reply by dillfrescott** [reply]
I'd also like to know how to modify the `payment-provider` value in a self hosted setup

**Reply by MrPorky** [reply]
> Hey, if you modify the `payment-provider` attribute to anything other than `free`, family plan will become accessible.
> 
> Please note that
> 
> 1. Family plan is designed to share your paid subscription, not your library. Each member gets their own private space. This feature might not be valuable in a self-hosted setup.
> 2. The best way to share your entire library would be to add the other user as a viewer / collaborator to all of your albums. There's a separate discussion tracking a "Share all" feature here: [Share all content with another user #787](https://github.com/ente-io/ente/discussions/787)

I would also love to change this. I feel the UI telling me I'm on a free plan is kind of weird 😄

**Reply by aleqx** [reply]
Where exactly is the `payment-provider` setting?

**Reply by andreasdueren** [reply]
Hey @vishnukvmd please let us know where we an modify the ```payment-provider```attribute. I can't find any reference to it in the codebase or wiki.

**Reply by mnvr** [reply]
When referring to setting the payment provider attribute, what Vishnu meant is that to initiate a family the user needs needs to be on a non-free plan. For this currently you'll need to modify the DB directly, see the subscriptions table. However, we'll make it easier! We'll modify museum that it doesn't require this check when self hosted.

Note that the check for non-free plan for the admin is only half of the story - the family management is via a separate web app - https://github.com/ente-io/families/ which also needs to be self hosted. When self hosting the families app, it should be modified to point to your self hosted museum. And your self hosted museum should be made aware of your families app (see apps.family in server/configurations/local.yaml)

https://github.com/ente-io/ente/blob/7eb4511402767925a89605ae8cee69589392d147/server/configurations/local.yaml#L104-L105

**Reply by mnvr** [reply]
> For this currently you'll need to modify the DB directly, see the subscriptions table. However, we'll make it easier! We'll modify museum that it doesn't require this check when self hosted.

This change is now done and merged in main. It will be part of the next docker image too.

---

All that said though, conceptually it doesn't really make sense to use the family plan when self hosting. The family plan is not for sharing photos, it is for sharing usage, and on your self hosted instance, you can use as much storage as you wish anyways.

(I guess what some people really want to change is not a family plan but instead show anything except "Free" in the plan description card when self hosting. That too can be done by creating a custom "product" and modifying the user's DB entry to use that custom product, but there isn't a one-liner for that currently)

**Reply by andreasdueren** [reply]
perfect, thank you!

> > For this currently you'll need to modify the DB directly, see the subscriptions table. However, we'll make it easier! We'll modify museum that it doesn't require this check when self hosted.
> 
> This change is now done and merged in main. It will be part of the next docker image too.
> 
> All that said though, conceptually it doesn't really make sense to use the family plan when self hosting. The family plan is not for sharing photos, it is for sharing usage, and on your self hosted instance, you can use as much storage as you wish anyways.
> 
> (I guess what some people really want to change is not a family plan but instead show anything except "Free" in the plan description card when self hosting. That too can be done by creating a custom "product" and modifying the user's DB entry to use that custom product, but there isn't a one-liner for that currently)
